### PR TITLE
fix(hasNotch): Missing device name for iPhone 13

### PR DESCRIPTION
**Pull request by rolfb**
## Description

Adds missing device name by code for iPhone 13 lineup. Relates to #1307

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅    |

### CODE REVIEW

**Review by schie — APPROVED** [review]
LGTM

File: ios/RNDeviceInfo/RNDeviceInfo.m [hunk 1]
```diff
@@ -240,11 +240,15 @@ - (NSDictionary *) getDeviceNamesByCode {
...
+	@"iPhone12,8": @"iPhone SE", // (2nd Generation iPhone SE),
         @"iPhone13,1": @"iPhone 12 mini",
         @"iPhone13,2": @"iPhone 12",
         @"iPhone13,3": @"iPhone 12 Pro",
         @"iPhone13,4": @"iPhone 12 Pro Max",
-        @"iPhone12,8": @"iPhone SE", // (2nd Generation iPhone SE)
+	@"iPhone14,1": @"iPhone 13 mini",
+        @"iPhone14,2": @"iPhone 13",
+        @"iPhone14,3": @"iPhone 13 Pro",
+        @"iPhone14,4": @"iPhone 13 Pro Max",
```
Comment (hiteshSavkare) [inline ios/RNDeviceInfo/RNDeviceInfo.m:251]: Hi @rolfb 
for iPhone 13 the above combination is not working , whereas for other devices of 13 series its working as expected , i tried with following combinations and looks me fine , i m not sure what is the number game of 1,2,34 vs 5,2,3,4 , sorry i am not IOS expert , just done some experiment with this numbers 
        @"iPhone14,5": @"iPhone 13 mini",
        @"iPhone14,2": @"iPhone 13",
        @"iPhone14,3": @"iPhone 13 Pro",
        @"iPhone14,4": @"iPhone 13 Pro Max",

Comment (rolfb) [reply ios/RNDeviceInfo/RNDeviceInfo.m:251]: @hiteshSavkare Great feedback, how exactly are you testing this?

Comment (hiteshSavkare) [reply ios/RNDeviceInfo/RNDeviceInfo.m:251]: @rolfb Thanks,
 react-native-device-info into is integrated in my project, so changed the file RNDeviceInfo.m with above code  and building the project with xcode 13 and testing on simulators of  13 series

Comment (rolfb) [reply ios/RNDeviceInfo/RNDeviceInfo.m:251]: @hiteshSavkare updated the file now, going to try to verify with the most recent xcode beta as well.

**Comment by rolfb** [comment]
@schie @mikehardy please hold back on merging this for a little while, need to verify codes with Xcode RC

File: ios/RNDeviceInfo/RNDeviceInfo.m (hunk 1 above)
Comment (hiteshSavkare) [reply ios/RNDeviceInfo/RNDeviceInfo.m:251]: Sounds good @rolfb,  thanks +1

**Comment by rolfb** [comment]
> @schie @mikehardy please hold back on merging this for a little while, need to verify codes with Xcode RC

Ok, I think we're good. Matching the exact values provided by Xcode RC itself from `Contents/Developer/Platforms/iPhoneOS.platform/usr/standalone/device_traits.db`

```
sqlite> select ProductType,ProductDescription from Devices where ProductDescription LIKE 'iPhone 13%';
ProductType|ProductDescription
iPhone14,2|iPhone 13 Pro
iPhone14,3|iPhone 13 Pro Max
iPhone14,4|iPhone 13 mini
iPhone14,5|iPhone 13
```

File: ios/RNDeviceInfo/RNDeviceInfo.m (hunk 1 above)
Comment (hiteshSavkare) [reply ios/RNDeviceInfo/RNDeviceInfo.m]: Great @rolfb , Hope you tested with simulators or Do i need to test again ?

Comment (rolfb) [reply ios/RNDeviceInfo/RNDeviceInfo.m]: @hiteshSavkare didn't check as I expect Xcode's own lookup tables to be correct. Feel free to verify as the list from Xcode isn't exactly matching the one you provided.

Comment (hiteshSavkare) [reply ios/RNDeviceInfo/RNDeviceInfo.m]: @rolfb looks fine :+1:

**Comment by mikehardy** [comment]
Thank you all so much!

**Comment by harisshk** [comment]
iPhone 13 series hasNotch() not working giving value as false

**Comment by mikehardy** [comment]
@harisshk https://stackoverflow.com/help/how-to-ask / https://stackoverflow.com/help/minimal-reproducible-example - I suspect you have outdated versions. Run our example from this repo on an iphone13 device and if you can reproduce open a new issue

**Comment by pckz** [comment]
@harisshk same for me, tested in the simulator

**Comment by schie** [comment]
@pckz what version are you on?

Like @mikehardy said to @harisshk, me thinks you're on an outdate version.

**Comment by pckz** [comment]
> @pckz what version are you on?
> 
> Like @mikehardy said to @harisshk, me thinks you're on an outdate version.

        "react-native": "0.62.2",
        "react-native-device-info": "^8.4.9",

(i've forced the latest version)

**Comment by mikehardy** [comment]
> same for me, tested in the simulator

Tested what in the simulator? Your app? Our example? Testing our example app would be best, no telling what your app is doing, since you haven't provided any details of it

**Comment by pckz** [comment]
> > same for me, tested in the simulator
> 
> Tested what in the simulator? Your app? Our example? Testing our example app would be best, no telling what your app is doing, since you haven't provided any details of it

My project runs:

    "react-native": "0.62.2",
    "react-native-device-info": "^8.4.9",

About the simulator, is Phone 13 running iOS 15.2. My XCode version is 13.2.1 (13C100)

I get *false* value doing this:

header :

     import DeviceInfo from 'react-native-device-info';

componentWillMount:
 
     console.log(DeviceInfo.hasNotch());

**Pull request merged**